### PR TITLE
Add Na_ontap_broadcast_domain module. 

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
We are updating and adding the Netapp ONTAP specific modules. The update modules add new options for the existing functions. The new modules allow for setting up and modify different part of the ONTAP system.

We have about 33 modules, we've been told to start with 2 or 3 first, and take feedback on these first before pushing them all out.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
na_ontap_broadcast_domain

##### ANSIBLE VERSION
```
ansible 2.6.0 (devel 0d1e9bfe90) last updated 2018/04/23 14:26:54 (GMT -400)
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /tmp/ansible/lib/ansible
  executable location = /tmp/ansible/bin/ansible
  python version = 2.7.5 (default, Oct 11 2015, 17:47:16) [GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by hulquest** [comment]
LGTM. shipit

File: lib/ansible/module_utils/netapp.py [hunk 1]
```diff
@@ -80,12 +80,22 @@ def has_sf_sdk():
...
 
     return dict(
         hostname=dict(required=True, type='str'),
         username=dict(required=True, type='str', aliases=['user']),
         password=dict(required=True, type='str', aliases=['pass'], no_log=True),
+        https=dict(required=False, type='str', default='False')
+    )
+
+
+def ontap_sf_host_argument_spec():
```
Comment (lmprice) [inline lib/ansible/module_utils/netapp.py]: ontap_sf, or na_sf?

Comment (carchi8py) [reply lib/ansible/module_utils/netapp.py]: this is the original function that works for the old CDOT, and SF. I'm setting this back to how it was before i had touched it. The function above (na_ontap_host_argument_spec) is the new one, for the new NA_ontap modules.

**Review by lmprice — APPROVED** [review]
LGTM shipit

**Comment by carchi8py** [comment]
Hey @mattclay, at yesterday core team meeting we were told we should be able to merge this in our self. I've check with @lmprice and @hulquest and my self to see if any of us could merge this in and we can't.  What our next step for merging this in?

File: lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: na_ontap_broadcast_domain
+short_description: Manage NetApp Ontap broadcast domains.
```
Comment (amit0701) [inline lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py]: Should be ONTAP

File: lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author: Chris Archibald (carchi8py@gmail.com), Kevin Hutton (khutton@netapp.com), Suhas Bangalore Shekar (bsuhas@netapp.com)
+description:
+- Modify a Ontap broadcast domain.
```
Comment (amit0701) [inline lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py]: Same here, should be ONTAP

File: lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+HAS_NETAPP_LIB = netapp_utils.has_netapp_lib()
+
+
+class NetAppOntapBroadcastDomain(object):
+    """
+        Create, Modifies and Destroys a Broadcast domain
+    """
+    def __init__(self):
+        """
+            Initialize the Ontap Net Route class
```
Comment (amit0701) [inline lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py]: Is this from  a different module? Should it be ONTAP broadcast domain class?

File: lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+                    else:  # execute modify
+                        self.modify_broadcast_domain()
+                elif self.state == 'absent':  # execute delete
+                    self.delete_broadcast_domain()
+        self.module.exit_json(changed=changed)
+
+
+def main():
+    """
+    Creates the NetApp Ontap Net Route object and runs the correct play task
```
Comment (amit0701) [inline lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py]: Should it be broadcast domain?

Comment (carchi8py) [reply lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py]: Ya looks like this was copied and pasted from another file let me fix that.

File: lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+        state=absent
+        username={{ netapp_username }}
+        password={{ netapp_password }}
+        hostname={{ netapp_hostname }}
+        broadcast_domain=123kevin
+        mtu=1000
+        ipspace=Default
+    - name: modify broadcast domain
+      na_ontap_broadcast_domain:
+        state=modify
```
Comment (amit0701) [inline lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py]: Why is the state modify?

File: lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+                    else:  # execute modify
+                        self.modify_broadcast_domain()
+                elif self.state == 'absent':  # execute delete
+                    self.delete_broadcast_domain()
+        self.module.exit_json(changed=changed)
+
+
+def main():
+    """
+    Creates the NetApp Ontap Broadcast Domain Object that can be created, deleted and modified.
```
Comment (amit0701) [inline lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py]: Might require this to be changed to ONTAP as well

File: lib/ansible/module_utils/netapp.py [hunk 8]
```diff
@@ -104,6 +114,34 @@ def create_sf_connection(module, port=None):
...
+    if HAS_NETAPP_LIB:
+        # set up zapi
+        server = zapi.NaServer(hostname)
+        server.set_username(username)
+        server.set_password(password)
+        if vserver:
+            server.set_vserver(vserver)
+        # Todo : Replace hard-coded values with configurable parameters.
+        server.set_api_version(major=1, minor=21)
+        # default is HTTPS
```
Comment (amit0701) [inline lib/ansible/module_utils/netapp.py]: Is it still true?

File: lib/ansible/module_utils/netapp.py [hunk 9]
```diff
@@ -104,6 +114,34 @@ def create_sf_connection(module, port=None):
...
+        if https in ['True', 'true']:
```
Comment (amit0701) [inline lib/ansible/module_utils/netapp.py]: Can this option be passed through the module? I don't see the parameter in the example below. Also can it be modified into a type="boolean" instead of type="str"

Comment (carchi8py) [reply lib/ansible/module_utils/netapp.py]: I think we had some issue when it was a boolean, but let me try again and see if it works.

Because we give a default on line 89 it only needs to be in an example where https is set to true.

**Review by amit0701 — APPROVED** [review]
LGTM Thanks Chris !
shipit

**Review by lmprice — APPROVED** [review]
shipit

**Comment by hulquest** [comment]
shipit

**Comment by carchi8py** [comment]
@jtanner @bcoca Hate to bother you guys, but we are not sure what the next steps are for us to merge this in.

We have a "shipit" label, so we thought we'd be able to merge, but it didn't happen automatically and we don't have a button to do it either. 

What the next steps to move this forward?

**Review by nitzmahone — APPROVED** [review]
